src/joinRates/JoinRates.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useCallback, useMemo } from 'react'
+
+import { Quality } from '../quality'
+import JoinRatesTable from './JoinRatesTable'
+import { JoinRatesProps } from './model'
+
+const JoinRates = ({
+  data,
+  onRowClickHandler,
+  selectedRules,
+  selectRules,
+  mode,
+  setMode,
+  pagination,
+  smallSize,
+}: JoinRatesProps): JSX.Element => {
+  const [_index, _setIndex] = useState<number>(0)
+  const _joinRatesDqData = useMemo(
+    () => (Array.isArray(data) ? data[_index].dq_result : data.dq_result),
```

Comment:
`data[_index].dq_result `=>  `data[_index]?.dq_result` 
In case `data=[]` we need to add `?` just in case